### PR TITLE
feat: refactor Roles Page

### PR DESCRIPTION
**Pull request by banushi-a**
Changes:
- Made necessary components client side (Navbar, roles page, etc.)
-  Screenshot 2024-04-09 at 10 03 44 PM
- The above full right preview thing is temporary since there isn't a design yet but its super easy to change
- Screenshot 2024-04-09 at 10 03 55 PM
- This page is missing the functionality of when you search, that specific company comes up. Def needs to be a different pr since we'll have to look into fuzzy search and test out edge cases

TBH I never looked over this code again so tear it up LOL

### CODE REVIEW

File: src/app/companies/page.tsx [hunk 1]
```diff
@@ -1,12 +1,32 @@
...
+import { RoleReviewCard } from "~/components/role-review-card";
 import SearchFilter from "~/components/search-filter";
+import { api } from "~/trpc/server";
+import { unstable_noStore as noStore } from "next/cache";
 
 export default async function Companies() {
+  /**
+   * FIXME: This is a temporary fix, figure out how to get build command working without noStore();
+   * @returns A promise containing the roles from the database
+   */
```
Comment (banushi-a) [inline src/app/companies/page.tsx:11]: One possible fix is to make this a client component and copy the structure of the roles/page.tsx to avoid this noStore issue. I'm definitely seeing (or imagining) some performance issues for this page

File: src/app/companies/page.tsx [hunk 2]
```diff
@@ -1,12 +1,32 @@
...
+
+  const roles = await getRoles();
+
   return (
-    <div className="flex h-full flex-col">
-      <HeaderLayout>
-        <SearchFilter />
-      </HeaderLayout>
-    </div>
+    <HeaderLayout>
```
Comment (RishikeshNK) [inline src/app/companies/page.tsx:21]: Can we move the header layout to the `layout.tsx` file? We can have multiple `layout.tsx` files to have different layouts for groups of pages (since some pages don't use the same header)?

See: [Pages and Layouts](https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts)

File: src/app/roles/page.tsx [hunk 3]
```diff
@@ -1,30 +1,45 @@
...
-          return <RoleReviewCard key={role.id} roleObj={role} />;
-        })}
-      </div>
+      {/* TODO: Loading animations */}
+      {reviews.data && (
+        <div className="mb-8 grid w-4/5 grid-cols-5 gap-4 lg:w-3/4">
+          <div className="col-span-2 gap-3">
+            {reviews.data.map((review, idx) => {
+              return (
+                <div key={review.id} onClick={() => setSelectedReviewIdx(idx)}>
```
Comment (RishikeshNK) [inline src/app/roles/page.tsx:25]: Do we want to store the index of the role being selected or the review object itself?

File: src/components/review-card-preview.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      className={cn(
+        "flex h-64 w-[100%] flex-col justify-between overflow-hidden rounded-3xl",
+        className,
+      )}
+    >
+      <div>
+        <CardHeader className="pb-3">
+          <div className="flex items-center justify-start space-x-4">
+            {company.data ? (
+              <img
```
Comment (RishikeshNK) [inline src/components/review-card-preview.tsx:51]: We can look into using `<Image />` later.

**Review by RishikeshNK — CHANGES_REQUESTED** [review]
Looks AMAZING! Again, some more nitpicks like last time that you can choose to ignore. LOVE IT! 🚀

**Review by cinyecai — APPROVED** [review]
ate !
